wagtail/admin/templates/wagtailadmin/shared/workflow_history/detail.html

```diff
@@ -1,153 +1,146 @@
-{% extends "wagtailadmin/base.html" %}
+{% extends "wagtailadmin/generic/base.html" %}
 {% load i18n wagtailadmin_tags %}
 
-{% block titletag %}{% trans "Workflow progress" %}{% endblock %}
+{% block main_content %}
+    <h2>{% icon object_icon classname="initial" %} {% latest_str object %}</h2>
```

Comment:
Previously this page doesn't have any header icon, but there's an icon beside the object's name:

<details>
<summary>Screenshot</summary>
<img width="374" alt="image" src="https://user-images.githubusercontent.com/6379424/224986671-3381f646-552a-42f3-9e7d-35e61c6bcbeb.png">
</details>

I feel like it looks a bit weird to have the same icon for both, so I used `list-ul` as the `header_icon` and came up with `object_icon` for the object's icon specifically for this view:

<details>
<summary>Screenshot</summary>
<img width="374" alt="image" src="https://user-images.githubusercontent.com/6379424/224986546-acb32d6d-c2d0-47b1-91c6-bf44e11ceb71.png">
</details>

I chose the `list-ul` icon because that's what we currently use for the Workflow information dialog:

<details>
<summary>Screenshot</summary>
<img width="615" alt="image" src="https://user-images.githubusercontent.com/6379424/224987337-1b9fd2ce-3f4d-4a51-a8c7-f86d84d18200.png">
</details>